src/server/routes/post_tx.rs

```diff
@@ -98,6 +102,22 @@ fn deep_hash_body(body: &PostTxBody) -> Result<Bytes, ValidatorServerError> {
     .map_err(|_| ValidatorServerError::InternalError)
 }
 
-async fn add_to_db(_body: &PostTxBody) -> Result<(), ValidatorServerError> {
-    Ok(())
+async fn add_to_db<Config>(ctx: &Config, body: &PostTxBody) -> Result<(), ValidatorServerError>
+where
+    Config: RequestContext + 'static,
+{
+    let tx = NewTransaction {
+        id: body.id.clone(),
+        epoch: 0,
```

Comment:
Shouldn't this be the current epoch? You can get the value through context: `server::routes::sign::Config::current_epoch(&self) -> i64`

That reminds me, the return value of this function is wrong, it should be `u128`.

Reply:
We discussed about this, currently the epoch should be left as i64 for Sqlite support. No problem

Reply:
Sorry, I was a bit unclear. You're correct, the database table should have `i64`, while the API I pointed out to, should have `u128`. The reason is that all domain logic should use correct types while the database layer should have the compromise. But the main point is that this value should not be set to zero.